### PR TITLE
feat(timeout-issue): link to custom GitHub host if user has set the environment variable

### PR DESCRIPTION
**Pull request by rmehner**
This is a first step into us being a bit more independent of the GitHub host.

### CODE REVIEW

File: lib/env.js [hunk 1]
```diff
@@ -1,6 +1,16 @@
 const envalid = require('envalid')
+const parseURL = require('url').parse
 const { str, url, num } = envalid
 
+if (process.env.GITHUB_HOST) {
+  try {
+    const parsed = parseURL(process.env.GITHUB_HOST)
+    process.env.GITHUB_URL = `${parsed.protocol}//${parsed.hostname}`
```
Comment (gr2m) [inline lib/env.js]: `hostname` does not include port number, maybe better to use `parsed.host`? Or are port numbers a non-issue  here?

File: test/content/timeout-issue.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  })
+
+  t.test('includes the link to the initial branch on the regular github host', async t => {
+    const content = requireFresh('../../content/timeout-issue')
+
+    const issueContent = content({fullName: 'finnp/abc'})
+    t.match(issueContent, /github\.com\/finnp\/abc/, 'includes the link to the repo at github.com')
+
+    t.end()
+  })
```
Comment (gr2m) [inline test/content/timeout-issue.js]: see my comment above, if there can be a case where `GITHUB_HOST` contains `port` numbers, lets add a test for it.

File: test/content/timeout-issue.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+test('timeout issue content', async t => {
+  t.afterEach(() => {
+    delete process.env.GITHUB_HOST
+    delete process.env.GITHUB_URL
+    cleanCache('../../lib/env')
+  })
+
+  t.test('includes the link to the initial branch on the custom github host', async t => {
+    process.env.GITHUB_HOST = 'https://enterprise.github/api/v3/'
```
Comment (gr2m) [inline test/content/timeout-issue.js]: I’m a bit confused that `GITHUB_HOST` has a path here (`/api/v3/`), maybe we can call it `GITHUB_API_BASE_URL`? Can we document the new environment variable somewhere?

Comment (rmehner) [inline test/content/timeout-issue.js]: It's not new, it existed before (https://github.com/greenkeeperio/greenkeeper/blob/46d0f51d54380a175ee68af3a96cf36dafa76453/lib/github.js#L36 & the test for it) ... that's why I've used it.

File: lib/env.js (hunk 1 above)
Comment (rmehner) [inline lib/env.js]: That's an excellent question, we should find out :)

**Comment by rmehner** [comment]
Rebased & used `host` instead of `hostname` ... anything else left?

**Pull request merged**